rand-core/src/impls.rs

```diff
@@ -333,6 +333,168 @@ impl<R: BlockRngCore + SeedableRng> SeedableRng for BlockRng<R> {
     }
 }
 
+
+
+/// Wrapper around PRNGs that implement [`BlockRngCore`] to keep a results
+/// buffer and offer the methods from [`RngCore`].
+///
+/// This is similar to [`BlockRng`], but specialized for algorithms that operate
+/// on `u64` values (rare). Hopefully specialization will one day allow
+/// `BlockRng` to support both, so that this wrapper can be removed.
```

Comment:
The "rare" part may change... I also don't know that it's worth talking about this future direction here. Also since all the implementations are different, is there any point making `BlockRng` support both?

Reply:
I'll remove the speculation...

The idea was once that things like `ReseedingRng` can then be used for both 32- and 64-bit RNGs.

Reply:
Which is a problem in itself. I wonder whether many people will complain about the new `ReseedingRng` being incompatible with many PRNGs... maybe not: [I'm struggling to find any references which are not simply copies of the source](https://github.com/search?p=1&q=ReseedingRng+language%3Arust&type=Code&utf8=%E2%9C%93)

Reply:
I found a grand total of two:
- [Gotham](https://github.com/gotham-rs/gotham/blob/master/gotham/src/middleware/session/rng.rs#L21)
- [Servo](https://github.com/servo/servo/blob/master/components/rand/lib.rs#L61)

Reply:
The servo one is little more than a reshuffle of Rand code, so more like one, and that's reseeding ChaCha, so we're fine.